plugins/movetoarchive.koplugin/main.lua

```diff
@@ -0,0 +1,161 @@
+    archive_dir_path = nil,
+    last_copied_from_dir = nil,
+}
+
+function MoveToArchive:init()
+    self.archive_dir_path = self.move_to_archive_settings:readSetting("archive_dir")
+    self.last_copied_from_dir = self.move_to_archive_settings:readSetting("last_copied_from_dir")
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function MoveToArchive:addToMainMenu(menu_items)
+    menu_items.move_to_archive = {
+        text = _("Move to archive"),
+        sub_item_table = {
+            {
+                text = _("Move current book to archive"),
+                callback = function() self:moveToArchive() end,
+                enabled_func = function() return self:isActionEnabled() end,
+            },
+            {
+                text = _("Copy current book to archive"),
+                callback = function() self:copyToArchive() end,
+                enabled_func = function() return self:isActionEnabled() end,
+            },
+            {
+                text = _("Go to archive folder"),
+                callback = function()
+                    if not self.archive_dir_path then
+                        self:showNoArchiveConfirmBox()
+                        return
+                    end
+                    self:openFileBrowser(self.archive_dir_path)
+                end,
+            },
+            {
+                text = _("Go to last copied/moved from folder"),
+                callback = function()
+                    if not self.last_copied_from_dir then
+                        UIManager:show(InfoMessage:new{
+                            text = _("No previous folder found.")
+                         })
+                        return
+                    end
+                    self:openFileBrowser(self.last_copied_from_dir)
+                end,
+            },
+            {
+                text = _("Set archive directory"),
+                keep_menu_open = true,
+                callback =  self.setArchiveDirectory,
+            }
+        },
+    }
+end
+
+function MoveToArchive:moveToArchive()
+    local move_done_text = _("Book moved.\nDo you want to open it from the archive folder?")
+    self:commonProcess(true, move_done_text)
+end
+
+function MoveToArchive:copyToArchive()
+    local copy_done_text =_("Book copied.\nDo you want to open it from the archive folder?")
+    self:commonProcess(false, copy_done_text)
+end
+
+function MoveToArchive:commonProcess(is_move_process, moved_done_text)
+    if not self.archive_dir_path then
+        self:showNoArchiveConfirmBox()
+        return
+    end
+    local document_full_path = self.ui.document.file
+    local filename
+    self.last_copied_from_dir, filename = util.splitFilePathName(document_full_path)
+
+    logger.dbg("MoveToArchive: last_moved/copied_from_dir :", self.last_copied_from_dir)
+    self.move_to_archive_settings:saveSetting("last_copied_from_dir", ("%s/"):format(self.last_copied_from_dir))
```

Comment:
Might need a `self.move_to_archive_settings:flush()` (as you did below in `MoveToArchive:setArchiveDirectory()`) to have it saved if you want to have that saving done each time.
_Or_ do that in a `MoveToArchive:onFlushSettings()` so it's done only once, when exiting a book.

Reply:
Done